src/ImmutableX.ts

```diff
@@ -350,6 +356,87 @@ export class ImmutableX {
       });
   }
 
+  /**
+   * Get a list of metadata refreshes
+   * @param ethSigner - the L1 signer
+   * @param collectionAddress - the Collection contract address
+   * @param pageSize - the page size of the result
+   * @param cursor - the cursor
+   * @returns a promise that resolves with the requested metadata refreshes
+   * @throws IMXError
+   */
+  public listMetadataRefreshes(
+    ethSigner: EthSigner,
+    collectionAddress?: string,
+    pageSize?: number,
+    cursor?: string,
+  ) {
+    return this.workflows
+      .listMetadataRefreshes(ethSigner, collectionAddress, pageSize, cursor)
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Get a list of metadata refresh errors
+   * @param ethSigner - the L1 signer
+   * @param refreshId - the metadata refresh ID
+   * @param pageSize - the page size of the result
+   * @param cursor - the cursor
+   * @returns a promise that resolves with the requested metadata refresh errors
+   * @throws IMXError
+   */
+  public getMetadataRefreshErrors(
+    ethSigner: EthSigner,
+    refreshId: string,
+    pageSize?: number,
+    cursor?: string,
+  ) {
+    return this.workflows
+      .getMetadataRefreshErrors(ethSigner, refreshId, pageSize, cursor)
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Get a list of metadata refresh results
+   * @param ethSigner - the L1 signer
+   * @param refreshId - the metadata refresh ID
+   * @returns a promise that resolves with the requested metadata refresh results
+   * @throws IMXError
+   */
+  public getMetadataRefreshResults(ethSigner: EthSigner, refreshId: string) {
+    return this.workflows
+      .getMetadataRefreshResults(ethSigner, refreshId)
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Request a metadata refresh
+   * @param ethSigner - the L1 signer
+   * @param request the request object containing the parameters to be provided in the API request
+   * @returns a promise that resolves with the requested metadata refresh ID
+   * @throws IMXError
+   */
+  public requestMetadataRefresh(
+    ethSigner: EthSigner,
+    request: CreateMetadataRefreshRequest,
+  ) {
+    return this.workflows
+      .requestMetadataRefresh(ethSigner, request)
+      .then(res => res.data.refresh_id)
```

Comment:
Is this correct? Do we want to return the `refresh_id` only as a string, or the whole `data` object? I think it makes sense to return the `CreateMetadataRefreshResponse`?

Reply:
Happy to change this. Currently that response only has refresh_id.
```
export interface CreateMetadataRefreshResponse {
    /**
     * The metadata refresh ID
     * @type {string}
     * @memberof CreateMetadataRefreshResponse
     */
    'refresh_id': string;
}
```

Reply:
I believe we currently return the whole response object despite number of attributes. Take [createProject](https://github.com/immutable/imx-core-sdk/blob/coresdk-interface-class/src/ImmutableX.ts#L360) for example, which only has one [attribute.](https://github.com/immutable/imx-core-sdk/blob/coresdk-interface-class/src/api/models/create-project-response.ts#L22)

Reply:
Hey @zatlite, this API call also returns a few headers which are related to their rate limit (how many refreshes remaining, and when their limit expires). Could those also please be returned here as part of the response body?

Reply:
@cajames We should discuss how the response headers should be treated. 

The code generator won't generate interfaces for the headers so users will always have to go through API references to figure out what fields are available. 
Is there a reason why they are in response headers instead of being part of the response body? 
Could they be moved into the body so that when the code is generated, SDK users can directly find the values from the response object instead of looking up the API reference doc to figure out what response headers are available?

We also noted that most of the rate limits (collection/mint) can be fetched from https://docs.x.immutable.com/reference/#/operations/getProject endpoint.

@Kastet @matthewmuscat

Reply:
Tracking the response headers in a new task DX-1249.